### PR TITLE
Refactored runner

### PR DESCRIPTION
**Pull request by EGAlberts**
A runner! for experiments. Please see the branch and its commits for more details. Also please test it out and try to break it, it is hard to find every edge case on my own. The shell scripts are in the docker folder so that during build they'd be copied into suave_ws.

### CODE REVIEW

**Review by Rezenders — COMMENTED** [review]
The README needs to be updated with instructions on how to use the runner.

I built the image from source, I changed a few things in the docker scripts. However, we still have that gazebo issue. To partially fix it I exported `GZ_VERSION="garden"` before building ros__gz, but now gz crashes when it is opened. Check: https://github.com/gazebosim/gz-sim/issues/1871 

I added a few commits to this PR. Can you check if you agree with the commits I added?

File: suave_metacontrol/suave_metacontrol/metacontrol_goal_update.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        self.pipeline_inspected = msg.data
+
+    
+    def goal_timer_cb(self):
+        if (not self.pipeline_detected and self.last_goal_sent != "generate_search_path"):
+            self.get_logger().info('\n\n Not Pipeline Detected and Not same goal as last time')
+            self.motion_future = self.send_adaptation_goal('maintain_motion')
+            self.get_logger().info('!!!Motion goal sent!!!')
+            adaptation_goal = "generate_search_path"
+        elif (self.pipeline_detected and not self.pipeline_inspected and self.last_goal_sent !="inspect_pipeline"):
```
[the remark this answers is not in the document]
Comment (Rezenders) [reply suave_metacontrol/suave_metacontrol/metacontrol_goal_update.py]: I fixed this

File: suave_metacontrol/suave_metacontrol/metacontrol_goal_update.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+            self.pipeline_detected_cb,
+            10, callback_group= subscriber_cb_group)
+
+        self.detect_subscription = self.create_subscription(
+            Bool,
+            'pipeline/inspected',
+            self.pipeline_inspected_cb,
+            10, callback_group= subscriber_cb_group)
+
+        self.mros_action_client = ActionClient(self, ControlQos, 'mros_objective', callback_group=client_cb_group)
```
[the remark this answers is not in the document]
Comment (Rezenders) [reply suave_metacontrol/suave_metacontrol/metacontrol_goal_update.py]: I fixed this

**Review by Rezenders — APPROVED** [review]

File: suave/launch/simulation.launch.py [hunk 3]
```diff
@@ -42,6 +42,15 @@ def generate_launch_description():
     x = LaunchConfiguration('x')
     y = LaunchConfiguration('y')
     z = LaunchConfiguration('z')
+    gui = LaunchConfiguration('gui')
+
+    gui_arg = DeclareLaunchArgument(
```
[the remark this answers is not in the document]
Comment (EGAlberts) [reply suave/launch/simulation.launch.py:47]: yes, remnant of that sorry :)

File: suave_missions/launch/mission.launch.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    adapt_manager_arg = DeclareLaunchArgument(
+        'adapt_manager',
+        default_value='none',
+        description='Which adaptation manager is in charge, none/metacontrol/random'
+    )
+
+    mission_type_arg = DeclareLaunchArgument(
+        'mission_type',
+        default_value='time_constrained_mission',
+        description='Which type of mission to have, time or distance'
```
[the remark this answers is not in the document]
Comment (EGAlberts) [reply suave_missions/launch/mission.launch.py:27]: yes, I think for now in the runner I complete this for them, so distance becomes distance_etc.etc.

File: suave_missions/suave_missions/const_dist_mission.py [hunk 5]
```diff
@@ -2,25 +2,28 @@
...
 from system_modes_msgs.srv import ChangeMode
-from suave.mission_planner import MissionPlanner
+
+from suave_missions.mission_planner import MissionPlanner
 
 
 class MissionConstDist(MissionPlanner):
     def __init__(self, node_name='const_dist_mission_node'):
         super().__init__(node_name)
-        self.mission_name = 'no adaptation - const distance'
```
[the remark this answers is not in the document]
Comment (EGAlberts) [reply suave_missions/suave_missions/const_dist_mission.py:15]: True, we should we add this

**Comment by Rezenders** [comment]
I should have opened a separate PR with the docker changes. Sorry about that.
